### DISCUSSION TITLE
Support __version__

### DISCUSSION THREAD

**Discussion by matthiasdiener** [illinois-ceesd/mirgecom#175] [Development] [answered]
```python
>>> import mirgecom
>>> mirgecom.__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'mirgecom' has no attribute '__version__'

# But:
>>> import numpy
>>> numpy.__version__
'1.19.4'

# And:
>>> from mirgecom import version
>>> version.VERSION_TEXT
'2020.1'
```

**Comment by inducer** [comment, marked as answer]
IMO: We should! In my packages, it's kind of hit-or-miss, often based on some bad practices that have become hard to change. :/ But in general `top_level_module.__version__` is the right place for a version number.

Side question: What's the accepted practice for comparing string-y version numbers? I know distutils/setuptools has something... is that what you're supposed to use? (That's why I have `VERSION` in my packages as a tuple-y version number, so that `VERSION >= (2020, 4)` does something vaguely sensible.)

**Reply by matthiasdiener** [reply]
> Side question: What's the accepted practice for comparing string-y version numbers? I know distutils/setuptools has something... is that what you're supposed to use? (That's why I have `VERSION` in my packages as a tuple-y version number, so that `VERSION >= (2020, 4)` does something vaguely sensible.)

I think https://www.python.org/dev/peps/pep-0440/ has something about it, and it seems to indicate distutils.

> Oh! Did not realize that this was "Discussions". Can this be turned into an issue?

Not as far as I can see unfortunately.

**Comment by inducer** [comment]
Oh! Did not realize that this was "Discussions". Can this be turned into an issue?
